pyseir/inference/infer_rt.py

```diff
@@ -468,6 +475,10 @@ def infer_all(self, plot=False, shift_deaths=0):
                     for col in df_all.columns:
                         if timeseries_type.value in col:
                             df_all[col] = df_all[col].shift(shift_in_days)
+                            # Extend death and hopitalization rt signals beyond shift to avoid sudden jumps in composit metric.
+                            df_all[col] = df_all[col].interpolate(
```

Comment:
some handy docs: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.interpolate.html

Reply:
I think we need to put more notes in here about how this call is doing double duty and being a little abused to do extrapolation too. 

For example, it does linear interpolation between two points (so 1, np.nan, 3 becomes 1,2,3) but it does forward fill for extrapolation (so 1, 2 , np.nan becomes 1, 2, 2). I think those behaviors are reasonable, but it's not clear from the code (or the docstring of the function) that it is following two separate paths.

So two options:
- Keep code as is and just add a more detailed comment
- Make it explicit by calling it interpolate twice with the 'limit_area' flag switched. This would also give us finer control over rules for extrapolation verse rules for interpolation. We could also set different 'limit' values for the max number of nan we are willing to replace. Depending on how the 'limit_area'='outer' is designed this might be a bit slower.

![image](https://user-images.githubusercontent.com/20585842/82594233-9e17ab00-9b58-11ea-8dc6-4ee625fbc03d.png)

Reply:
Oh, nice catch Brett. Is that the behavior we want for forward extrapolation? I think it would be better to have it do some version of a linear extrapolation...

Also, just pinning I can imagine edge cases where linearly interpolating between these points won't model the true distribution well, but I don't think we should try to deal with that now.

Reply:
But also, if while we are adding information, are you back-extrapolating? (e.g. when the death series starts after the case series and we see a sudden uptick in the inferred Rt when the death series 'begins')

Reply:
[Rt_results__Kansas__20.pdf](https://github.com/covid-projections/covid-data-model/files/4664432/Rt_results__Kansas__20.pdf)

Reply:
I'll make it clearer exactly what the behavior is here. I evaluated other ways of extrapolating the end of the series but didn't see anything that was obviously better.